### PR TITLE
[4/4] Remove `TensorBuffer::JPEG`, `DecodedTensor`, `TensorDecodeCache`

### PR DESCRIPTION
**Pull request by emilk**
* Part of https://github.com/rerun-io/rerun/issues/6844
* Closes https://github.com/rerun-io/rerun/issues/3803

⚠️ This breaks any existing `JPEG`-encoded RRDs

### Rust API
* Removed `TensorBuffer::JPEG`
* Removed `TensorData::from_jpeg_bytes`
* Deprecated `Image::from_file_path` and `from_file_contents`

For all of these, use `ImageEncoded` instead.

### PR train
* Prev: https://github.com/rerun-io/rerun/pull/6882
* Prev: https://github.com/rerun-io/rerun/pull/6874
* Prev: https://github.com/rerun-io/rerun/pull/6883

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6884?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6884?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/6884)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_data_ui/src/image.rs [hunk 1]
```diff
@@ -393,19 +377,13 @@ pub fn tensor_summary_ui_grid_contents(
         | TensorBuffer::F16(_)
         | TensorBuffer::F32(_)
         | TensorBuffer::F64(_) => {}
-        TensorBuffer::Jpeg(jpeg_bytes) => {
```
Comment (Wumpf) [inline crates/viewer/re_data_ui/src/image.rs:396]: this makes me realize that there's no custom data ui for `EncodedImage` yet unless I missed a blob of code along the way.
Needs to be put on the todo list! 

Edit: Ah this is a much deeper problem obviously. There's no `EncodedImage` component, so this whole line of work makes us loose the ability to see images in the property view. We may want to come up with something new here short or long term. Worth discussing today or later this week :)
image

**Review by Wumpf — APPROVED** [review]
looking great!

**Comment by emilk** [comment]
@rerun-bot full-check

**Comment by emilk** [comment]
@rerun-bot full-check

**Pull request merged**